contracts/interfaces/IStakingPool.sol

```diff
@@ -18,6 +18,8 @@ struct AllocationRequest {
   uint capacityReductionRatio;
   uint rewardRatio;
   uint globalMinPrice;
+  uint32 newPeriod;
+  uint32 remainingPeriod;
```

Comment:
An alternative approach here is the following:

add only 1 new field called extraPeriod

let AllocationRequest.period = be the entire new period whether it's a new cover or an edit

In such a way that if it's an edit:

remainingPeriod = allocationRequest.period - allocationRequest.extraPeriod